### DISCUSSION TITLE
Import From .env File

### DISCUSSION THREAD

**Discussion by ManojKiranA** [envault/envault#20] [Ideas]
It woluld be nice if we have option to import all Key Value pairs from .env File

**Comment by danharrin** [comment]
Hey @ManojKiranA, this feature actually exists already 😄

To use it, you can click the "Import" button in the bottom left corner of the "New variable" dialog.

https://user-images.githubusercontent.com/41773797/103437681-306d4a80-4c22-11eb-88bb-41ce09a1ec2a.mov

**Comment by ManojKiranA** [comment]
I think you misunderstood slightly. Instead of copying and pasting the content of .env file into import Textbox. I was suggesting for a feature where we need to select a .env directly and import all the Key Value pairs from it. I hope you understand it now

**Reply by ManojKiranA** [reply]
> I think you misunderstood slightly. Instead of copying and pasting the content of .env file into import Textbox. I was suggesting for a feature where we need to select a .env directly and import all the Key Value pairs from it. I hope you understand it now

I hope now you got my Feature idea right ??

**Reply by danharrin** [reply]
Ah, so you would like to upload a .env file and it gets extracted?

**Reply by danharrin** [reply]
I actually added an alternative idea to the Roadmap - being able to upload .env values from the CLI.

**Reply by danharrin** [reply]
This feature is now available in [v2.4.0](https://github.com/envault/envault/releases/tag/v2.4.0).

**Reply by danharrin** [reply]
Here's a demo:

https://user-images.githubusercontent.com/41773797/104010820-05947080-51a5-11eb-8631-64e6db6e1553.mp4

**Comment by ManojKiranA** [comment]
Meanwhile when i try to import Multiple variables Its not working as expexted. After some digging i have found [here](https://github.com/envault/envault/blob/master/app/Http/Livewire/Variables/Create.php#L47) that PHP_EOL is [not working properly](https://stackoverflow.com/a/5508461) in textarea. May be regex can fix the issue i think so

https://user-images.githubusercontent.com/30294553/103545609-cb3a8300-4ec7-11eb-9fca-a255c3de6eb0.mp4

**Comment by danharrin** [comment]
Hey @ManojKiranA! This looks like a bug, thanks for highlighting it.

If you are able to, creating a pull request with a fix would be much appreciated.

**Comment by danharrin** [comment]
This issue has been fixed by [v2.1.0](https://github.com/envault/envault/releases/tag/v2.1.0).
